cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/FITSCutout.java

```diff
@@ -0,0 +1,102 @@
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fits.slice;
+
+import ca.nrc.cadc.dali.DaliUtil;
+import ca.nrc.cadc.wcs.exceptions.NoSuchKeywordException;
+import ca.nrc.cadc.wcs.exceptions.WCSLibRuntimeException;
+import nom.tam.fits.Header;
+import nom.tam.fits.HeaderCardException;
+
+public abstract class FITSCutout<T> {
+    protected final FITSHeaderWCSKeywords fitsHeaderWCSKeywords;
+
+    public FITSCutout(final Header header) throws HeaderCardException {
+        DaliUtil.assertNotNull("header", header);
+        this.fitsHeaderWCSKeywords = new FITSHeaderWCSKeywords(header);
+    }
+
+    protected FITSCutout(final FITSHeaderWCSKeywords fitsHeaderWCSKeywords) {
+        DaliUtil.assertNotNull("fitsHeaderWCSKeywords", fitsHeaderWCSKeywords);
+        this.fitsHeaderWCSKeywords = fitsHeaderWCSKeywords;
+    }
+
+
+    /**
+     * Obtain the bounds of the given cutout.
+     * @param cutoutBound   The bounds (shape, interval etc.) of the cutout.
+     * @return  long[] array of overlapping bounds, or long[0] if all pixels are included.
```

Comment:
Does this return null if no pixels are included? The caom2 cutout code had those 3 return options... or something else?

Reply:
Returns null, like CAOM-2 does.